src/lucky_web/tags/text_helpers.cr

```diff
@@ -0,0 +1,217 @@
+module LuckyWeb::TextHelpers
+  @@_cycles = Hash(String, Cycle).new
+
+  def truncate(text, length = 30, omission = "...", separator = nil, escape = false, blk : Nil | Proc = nil)
+    if text
+      content = truncate_text(text, length, omission, separator)
+      content = escape ? HTML.escape(content) : content
+      content += blk.call.to_s if blk.is_a?(Proc) && text.size > length
+      content
+    end
+  end
+
+  def truncate(text, length = 30, omission = "...", separator = nil, escape = true, &block : -> _)
+    truncate(text, length, omission, separator, escape, blk: block)
+  end
+
+  # This could go in a String extension
+  private def truncate_text(text, truncate_at, omission = "...", separator = nil)
+    return text unless text.size > truncate_at
+
+    length_with_room_for_omission = truncate_at - omission.size
+    stop = \
+      if separator
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text, phrases : Nil | String | Regex | Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
+    if text.to_s.blank? || phrases.to_s.blank?
+      text.nil? ? "" : text
+    else
+      unless phrases.is_a?(Array)
```

Comment:
Instead of allow non-arrays, I think we can require an array. That way this code is unnecessary and we have less tests. I also think it might read a bit better that way: `phrases: "a phrase"` seems a bit strange because `phrases` is plural, but you're passing in just one object.